src/System.Net.WebClient/src/System/Net/WebClient.cs

```diff
@@ -1176,6 +1176,101 @@ private string MapToDefaultMethod(Uri address)
                 "STOR" :
                 "POST";
         }
+        
+        private static string UrlEncode(string str)
+        {
+            if (str == null)
+                return null;
+            byte[] bytes = Encoding.UTF8.GetBytes(str);
+            return Encoding.ASCII.GetString(UrlEncodeBytesToBytesInternal(bytes, 0, bytes.Length, false));
+        }
+
+        private static byte[] UrlEncodeBytesToBytesInternal(byte[] bytes, int offset, int count, bool alwaysCreateReturnValue)
+        {
+            int cSpaces = 0;
+            int cUnsafe = 0;
+
+            // count them first
```

Comment:
nit: Pls follow current convention for comments in networking code. Uppercase first word and end w/ period.